app/controllers/buyers/users_controller.rb

```diff
@@ -2,29 +2,45 @@ class Buyers::UsersController < Buyers::BaseController
   activate_menu :audience, :accounts, :listing
 
   inherit_resources
-  actions :index, :show, :edit, :update, :destroy
+  actions :index, :show, :edit
   defaults :route_prefix => 'admin_buyers' #FIXME: inherited_resource makes us repeat this
   belongs_to :account, :collection_name => :buyer_accounts
 
   def update
     # TODO: I think this controller is used only on provider side
-    resource.validate_fields! if current_user.account.buyer?
-    resource.attributes = params[:user]
-    super
+    resource.validate_fields! if current_account.buyer?
+    resource.attributes = params[:user] # TODO: we should filter the params
+
+    if update_resource(resource, resource_params)
```

Comment:
Can we do a normal `resource.update_attributes` and start getting rid of `InheritResources`?

Reply:
@guicassolato Unfortunately, that wouldn't work for updating the `role`, even with keeping the previous `resource.attributes = params[:user]`